pkg/imgpkg/imagetar/tar_reader.go

```diff
@@ -28,6 +32,93 @@ func (r TarReader) Read() ([]imagedesc.ImageOrIndex, error) {
 	return imagedesc.NewDescribedReader(ids, file).Read(), nil
 }
 
+// PresentLayers retrieves all the layers that are present in a tar file
+func (r TarReader) PresentLayers() ([]v1.Layer, error) {
+	var result []v1.Layer
+	allImages, err := r.Read()
+	if err != nil {
+		return nil, err
+	}
+	for _, image := range allImages {
+		if image.Image != nil {
+			img := *image.Image
+			layers, err := r.presentLayersForImage(img)
+			if err != nil {
+				return nil, fmt.Errorf("Processing Image %s: %s", image.OrigRef, err)
+			}
+			result = append(result, layers...)
+		} else if image.Index != nil {
+			idx := *image.Index
+			layers, err := r.presentLayersForIndex(idx)
+			if err != nil {
+				return nil, fmt.Errorf("Processing Index %s: %s", image.OrigRef, err)
+			}
+			result = append(result, layers...)
+		}
+	}
+
+	return result, nil
+}
+
+func (r TarReader) presentLayersForImage(img v1.Image) ([]v1.Layer, error) {
+	var result []v1.Layer
+	layers, err := img.Layers()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to retrieve layers: %s", err)
+	}
+
+	for _, layer := range layers {
+		h, err := layer.Digest()
+		if err != nil {
+			return nil, fmt.Errorf("Unable to get digest from layer: %s", err)
+		}
+		r, err := layer.Compressed()
+		if err != nil {
+			continue
+		}
+
+		size, err := layer.Size()
+		if err != nil {
+			return nil, err
+		}
+		closer, err := verify.ReadCloser(r, size, h)
+		if err != nil {
+			return nil, err
+		}
+
+		n, err := io.Copy(io.Discard, closer)
+		if err != nil || n == 0 {
```

Comment:
under what conditions `n == 0` happens vs err!=nil?

Reply:
(not sure which case we are trying to address with n==0)

Reply:
Not sure if this was the result of a prior error I was getting when calling the reader or not. So I just kept it. But there were cases where it would not fail and read 0 bytes, nevertheless it might not be the case anymore since I fixed something else. I will try to remove it to see if it works without it.

Reply:
I just run the tests locally and it look like that we can remove the `n==0` portion